WalletWasabi/WabiSabi/Client/ArenaClient.cs

```diff
@@ -64,6 +65,34 @@ public ArenaClient(WabiSabiClient amountCredentialClient, WabiSabiClient weightC
 			return inputRegistrationResponse.AliceId;
 		}
 
+		public async Task RemoveInputAsync(Guid roundId, Guid aliceId)
+		{
+			await RequestHandler.RemoveInputAsync(new InputsRemovalRequest(roundId, aliceId)).ConfigureAwait(false);
+		}
+
+		public async Task RegisterOutputAsync(Guid roundId, TxOut output, IEnumerable<Credential> amountCredentialsToPresent, IEnumerable<Credential> weightCredentialsToPresent)
+		{
+			var presentedAmount = amountCredentialsToPresent.Take(2).Sum(x => (long)x.Amount.ToUlong());
```

Comment:
What is this magic number two? Can it be less or more? 
The number of amountCredentialsToPresent should be qual of the number of amountCredentialsToPresent?

Reply:
This magic `2` is the `NumberOfCredentials` and yes, it has to be removed from here and moved to the `WabiSabiClient`. It is in the top one of my TO DO list and my idea is to do it after your #5488 is merged but I can do it as part of this PR is your prefer.

Reply:
`Guard` asserting that `amountCredentialsToPresent.Count()` equals the parameter from #5488 and then just `Sum` the entire collection is what I would suggest

Reply:
Done. The `WabiSabiClient` is smart enough to handle all situations where we provide less credentials but we should check not to have more than required credentials.